### ISSUE TITLE
Create standalone Net Core App

### ISSUE THREAD

**Issue by jaredpar** [dotnet/sdk#1138]
How can I setup my project so that the build output has a fully runnable .Net Core application?  I don't want it to be dependent on the shared framework being installed.  Hence all dependencies should appear in the output directory.

**Comment by nguerrera** [comment]
There is not currently a way to do this straight to the output directory. You need to use the publish verb/target and specify the runtime identifier that you want your standalone publish output to run on.

dotnet publish -r [RID]
msbuild /t:publish /p:RuntimeIdentifier=[RID]

https://docs.microsoft.com/en-us/dotnet/articles/core/deploying
https://docs.microsoft.com/en-us/dotnet/articles/core/rid-catalog
https://docs.microsoft.com/en-us/dotnet/articles/core/deploying/deploy-with-cli

**Comment by jaredpar** [comment]
@nguerrera tried that out and it got me a bit further.  All of the dependencies are deployed but no exe.  In the past we were using CoreRun as the kick off executable but can't find an equivalent here.  How do I run the output of `/t:Publish`?

**Comment by nguerrera** [comment]
If your project has OutputType=Exe, you'll get an  exe with your assembly name that runs your app.

**Comment by jaredpar** [comment]
@nguerrera not seeing that.  Verified my OutputType is EXE.

**Comment by nguerrera** [comment]
Can you point me to your branch and repro steps?

**Comment by nguerrera** [comment]
I think this is probably because you can't reference Microsoft.NETCore.App. I'll figure out a workaround and it will help to have your actual setup to confirm.

**Comment by jaredpar** [comment]
@nguerrera sure

1. Git clone https://github.com/jaredpar/roslyn
1. Switch branch to fix-correct
1. Run Restore.cmd in the root
1. cd `src\Tools\Source\CompilerGeneratorTools\Source\CSharpErrorFactsGenerator`
1. Run `msbuild /v:m`
1. Run `msbuild /t:publish`

The output folder will be `Binaries\Debug\Exes\CSharpErrorFactsGenerator\netcoreapp1.1\publish` relative to the clone directory.

**Comment by jaredpar** [comment]
@nguerrera that particular project can reference Microsoft.NetCore.App.  Tried it though and it didn't fix the issue.

**Comment by nguerrera** [comment]
Weird. Taking a look.

**Comment by jaredpar** [comment]
@nguerrera came by and we eventually figured this out.  The reason it wasn't working for me came down to a couple of problems

- The project file 
    - Did not have an explicit `<RuntimeIdentifier>` entry. 
    - Did have old legacy entries that likely needed to be removed: `<NuGetRuntimeIdentifier>`, `<RuntimeIndentifier>` (intentionally misspelled) 
- The project did not reference `Microsoft.NETCore.App`.  This is needed to bring in the `dotnet.exe` executable.  

Once we made those changes, and worked through a couple PEBKAC problems, I was able to get everything squared away.
